src/state/slices/stakingDataSlice/stakingDataSlice.test.ts

```diff
@@ -0,0 +1,522 @@
+            'cosmosvaloper1r9lxkpqre6j4487ut882xchgr7rdtx3x76gtdp',
+            'cosmosvaloper1yvwqd5rdtuaw25mcqhz794dvgq9k9yeh8mjcdh'
+          ])
+        })
+      })
+    })
+
+    describe('selectSingleValidator', () => {
+      it('returns null when state is empty', async () => {
+        const selected = selectSingleValidator(
+          store.getState(),
+          cosmosAccountSpecifier,
+          SHAPESHIFT_VALIDATOR_ADDRESS
+        )
+        expect(selected).toBeNull()
+      })
+
+      describe('validators and staking data are loaded', () => {
+        beforeEach(() => {
+          store.dispatch(
+            stakingData.actions.upsertValidatorData({
+              validators: mockValidatorData
+            })
+          )
+        })
+
+        it('returns null when validator address is unknown', async () => {
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+
+          const selected = selectSingleValidator(
+            store.getState(),
+            cosmosAccountSpecifier,
+            'cosmosvaloper1xxjvzwjpsf6ktuffkcpx29ut9qfrn0my8xdtqd'
+          )
+          expect(selected).toBeNull()
+        })
+
+        it('returns validator info when validator address is known', async () => {
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+
+          const selected = selectSingleValidator(
+            store.getState(),
+            cosmosAccountSpecifier,
+            SHAPESHIFT_VALIDATOR_ADDRESS
+          )
+          expect(selected).toEqual({
+            address: 'cosmosvaloper199mlc7fr6ll5t54w7tts7f4s0cvnqgc59nmuxf',
+            apr: '0.1496681491',
+            commission: '0.100000000000000000',
+            moniker: 'ShapeShift DAO'
+          })
+        })
+      })
+    })
+
+    describe('selectRewardsAmountByAssetId', () => {
+      it('returns empty string when state is empty', async () => {
+        const selected = selectRewardsAmountByAssetId(
+          store.getState(),
+          cosmosAccountSpecifier,
+          SHAPESHIFT_VALIDATOR_ADDRESS,
+          cosmosAssetId
+        )
+        expect(selected).toEqual('')
+      })
+
+      describe('validators and staking data are loaded', () => {
+        beforeEach(() => {
+          store.dispatch(
+            stakingData.actions.upsertValidatorData({
+              validators: mockValidatorData
+            })
+          )
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+        })
+
+        it('returns empty string when there is no reward', async () => {
```

Comment:
No rewards != no rewards data, you could have rewards on-chain but for xyz reason related to our business logic the store isn't populated with it
```suggestion
        it('returns empty string when there is no rewards data', async () => {
```